tsdb/head_read.go

```diff
@@ -319,27 +327,25 @@ func (s *memSeries) chunk(id chunks.HeadChunkID, chunkDiskMapper *chunks.ChunkDi
 	// The max index for the s.mmappedChunks slice can be len(s.mmappedChunks)-1, hence if the ix
 	// is len(s.mmappedChunks), it represents the next chunk, which is the head chunk.
 	ix := int(id) - int(s.firstChunkID)
-	if ix < 0 || ix > len(s.mmappedChunks) {
+	if ix < 0 || ix > len(s.mmappedChunks)+len(s.headChunks) {
```

Comment:
`ix` is 0 based. Previously `len(s.mmappedChunks)` meant the last head chunk. Now `len(s.mmappedChunks)+len(s.headChunks)-1` is the last chunk.
```suggestion
	if ix < 0 || ix > len(s.mmappedChunks)+len(s.headChunks)-1 {
```

Because tests didn't fail, we need to add a test case such that `return s.headChunks[ix], false, nil` would have panicked.